randovania/generator/generator_reach.py

```diff
@@ -77,6 +79,17 @@ def nodes(self) -> Iterator[Node]:
     def safe_nodes(self) -> Iterator[Node]:
         raise NotImplementedError
 
+    @property
+    def safe_uncollected_resource_nodes(self) -> Iterator[Node]:
+        context = self.node_context()
+        for node in self.safe_nodes:
+            if node.is_resource_node:
+                res = typing.cast(ResourceNode, node)
+                if res.should_collect(self.node_context()) and res.requirement_to_collect().satisfied(
```

Comment:
Re-use the `context` variable from above

Reply:
LMAO that's literally why i added it whoops